ros/src/tl_detector/tl_detector.py

```diff
@@ -68,9 +93,19 @@ def image_cb(self, msg):
             msg (Image): image from car-mounted camera
 
         """
+        # Do not use every frame for classification, but every nth
+        self.frame_nb += 1
+        if self.frame_nb % self.use_tf_nth_frame == 0:
+            self.frame_nb = 0
+        else:
+            return
+
         self.has_image = True
         self.camera_image = msg
         light_wp, state = self.process_traffic_lights()
+        readable_state = self.state_to_text(state)
+
+        rospy.logwarn("Closest light wp: {0} \n And light state: {1}".format(light_wp, readable_state))
```

Comment:
Logging can really slow things down, try commenting it out for better performance. It mattered for the waypoint updater at least, but it might have been running with higher frequency.

Reply:
Thats a good point!

Reply:
Logging will be disabled before merge into master.